webpush/encrypt.go

```diff
@@ -139,36 +142,64 @@ type EncryptionResult struct {
 	ServerPublicKey []byte
 }
 
+// Decrypt an encrypted messages.
+func Decrypt(sub *Subscription, crypt *EncryptionResult, subPrivate []byte) (plain []byte, err error) {
+	secret := sharedSecret(curve, crypt.ServerPublicKey, subPrivate)
+	prk := hkdf(sub.Auth, secret, authInfo, 32)
+
+	// Derive the Content Encryption Key and nonce
+	ctx := newContext(sub.Key, crypt.ServerPublicKey)
+	cek := newCEK(ctx, crypt.Salt, prk)
+	nonce := newNonce(ctx, crypt.Salt, prk)
+
+	plain, err = decrypt(crypt.Ciphertext, cek, nonce)
+	if err != nil {
+		return nil, err
+	}
+	return
+}
+
+// TODO: input should be a []byte ( proto, etc )
```

Comment:
I agree, but it's probably better to raise a GitHub issue rather than add new TODOs.

(I know I already have a couple in the code that I should totally get rid of)